### PR TITLE
fix: feed load

### PR DESCRIPTION
**Pull request by rebelchris**
## Changes

This PR cleans up the initial feed request.
Before we would load pretty much ALL data we could ever need on a post, but thinking about it, it's actually more rare for users to need all these details (which include many joins etc on database as well)

So we decided to move to rather load the full post data once the users needs it:
- Opens context menu
- Open grid modal

It's hard to determine what's changed but here are some stats: (TODO, Need deployment URL)
| Before | After |
|--------|--------|
| Cell | Cell |
| Cell | Cell | 

## REQUEST ⏰ 

Please do test it out on preview if you can break any modal :) (might have missed some edge-case data option)

## Events

Did you introduce any new tracking events?

## Experiment

Did you introduce any new experiments?

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

### CODE REVIEW

File: packages/shared/src/graphql/posts.ts [hunk 1]
```diff
@@ -189,6 +189,7 @@ export const POST_BY_ID_QUERY = gql`
       trending
       content
       contentHtml
+      pinnedAt
```
Comment (nensidosari) [inline packages/shared/src/graphql/posts.ts:192]: Does this fix this issue? https://dailydotdev.slack.com/archives/C01L0QXQJNB/p1729159495038119

**Review by nensidosari — APPROVED** [review]
Looks great!

File: packages/shared/src/components/post/withPostById.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    const { post } = props;
+
+    const { post: loadedPost, isLoading } = usePostById({
+      id: post?.id,
+      options: { initialData: { post } },
+    });
+
+    if (!post.id || isLoading) {
+      return null;
+    }
```
Comment (capJavert) [inline packages/shared/src/components/post/withPostById.tsx:30]: Should we return `null` or was this temporary to prevent rendering any component that might use something we removed?

File: packages/shared/src/components/post/withPostById.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+    return <WrappedComponent {...props} post={loadedPost} />;
+  };
+
+  WithPostById.displayName = 'WithFeaturesBoundary';
```
Comment (capJavert) [inline packages/shared/src/components/post/withPostById.tsx:35]:
```suggestion
  WithPostById.displayName = 'WithPostById';
```

**Review by capJavert — COMMENTED** [review]
Nice work!

File: packages/shared/src/graphql/posts.ts (hunk 1 above)
Comment (rebelchris) [reply packages/shared/src/graphql/posts.ts:192]: No I decided not to fix it here, it probably needs some bigger refactoring 🙏

File: packages/shared/src/components/post/withPostById.tsx (hunk 2 above)
Comment (rebelchris) [reply packages/shared/src/components/post/withPostById.tsx:30]: @capJavert We should not continue as the render needs the post basically, it's just to prevent the initial render which is still null to not break the full UI.
It's pretty instant due to the initial data though.

Comment (capJavert) [reply packages/shared/src/components/post/withPostById.tsx:30]: Ah ok so it goes through if `initialData` is passed? Thats ok then 👌

File: packages/shared/src/hooks/usePostById.ts [hunk 4]
```diff
@@ -169,7 +172,7 @@ const usePostById = ({ id, options = {} }: UsePostByIdProps): UsePostById => {
       post: post?.post,
       relatedCollectionPosts: post?.relatedCollectionPosts,
       isError,
-      isLoading,
+      isLoading: !post?.post ?? isLoading,
```
Comment (rebelchris) [inline packages/shared/src/hooks/usePostById.ts:175]: This is a bit weird, but we also not loading is we have initialData basically (from the actual post return)

File: packages/shared/src/components/Feed.tsx [hunk 5]
```diff
@@ -386,17 +373,17 @@ export default function Feed<T>({
...
+  //   onBookmark({
+  //     post,
+  //     origin,
+  //     opts: {
+  //       row,
+  //       column,
+  //       columns: virtualizedNumCards,
+  //       ...feedLogExtra(feedName, ranking),
+  //     },
+  //   });
```
Comment (rebelchris) [inline packages/shared/src/components/Feed.tsx:386]:
```suggestion
```

File: packages/shared/src/hooks/vote/types.ts [hunk 6]
```diff
@@ -92,8 +92,9 @@ export const voteMutationMatcher: UseMutationMatcher<
 
   return (
     status === 'success' &&
-    mutation?.options?.mutationKey?.toString() ===
-      createVoteMutationKey({ entity }).toString()
+    mutation?.options?.mutationKey
```
Comment (rebelchris) [inline packages/shared/src/hooks/vote/types.ts:95]: @capJavert I decided to rather check for includes as the feed one itself has something like:
`'post', 'mutation', 'vote', 'my-feed'` where the page one only has: `'post', 'mutation', 'vote'`

This way they both hit <-> both weys

File: packages/shared/src/hooks/vote/useFeedVotePost.ts [hunk 7]
```diff
@@ -43,14 +43,6 @@ export const useFeedVotePost = ({
...
     variables: { feedName },
-    onMutate: ({ id, vote }) => {
-      return mutateVoteFeedPost({
-        id,
-        vote,
-        items,
-        updatePost,
-      });
-    },
   });
```
Comment (rebelchris) [inline packages/shared/src/hooks/vote/useFeedVotePost.ts:46]: @capJavert Because of set message above we actually don't need this here, we can just rely on the mutationSubscription for the feed ones

File: packages/shared/src/components/PostOptionsMenu.tsx [hunk 8]
```diff
@@ -105,6 +105,13 @@ export default function PostOptionsMenu({
     useContextMenu({
       id: contextId,
     });
+  const { post: loadedPost } = usePostById({
+    id: initialPost?.id,
+    options: {
+      initialData: useCallback(() => ({ post: initialPost }), [initialPost]),
+    },
+  });
+  const post = loadedPost ?? initialPost;
```
Comment (capJavert) [inline packages/shared/src/components/PostOptionsMenu.tsx:114]: Do we need `initialData` then?

File: packages/shared/src/components/post/withPostById.tsx [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+): WrappedComponentType<Props, LayoutProps> => {
+  const WithPostById = (props: Props & { post?: Post }): ReactElement => {
+    const { post } = props;
+
+    const { post: loadedPost, isLoading } = usePostById({
+      id: post?.id,
+      options: { initialData: useCallback(() => ({ post }), [post]) },
+    });
+
+    if (!post.id || isLoading) {
```
Comment (capJavert) [inline packages/shared/src/components/post/withPostById.tsx:33]:
```suggestion
    if (!post?.id || isLoading) {
```

File: packages/shared/src/hooks/usePostById.ts (hunk 4 above)
Comment (capJavert) [reply packages/shared/src/hooks/usePostById.ts:175]: `!post?.post` will always be boolean so `??` will never evaluate, I think it needs:
```suggestion
      isLoading: !post?.post || isLoading,
```

File: packages/shared/src/hooks/vote/useFeedVotePost.ts (hunk 7 above)
Comment (capJavert) [reply packages/shared/src/hooks/vote/useFeedVotePost.ts:46]: One thing that this changes though is that upvote will not show optimistically until server side call actually passes as to on current production upvote will instantly turn green on click.

**Review by capJavert — APPROVED** [review]
Looks good now, few caveats, but not blocking

File: packages/shared/src/components/PostOptionsMenu.tsx (hunk 8 above)
Comment (rebelchris) [reply packages/shared/src/components/PostOptionsMenu.tsx:114]: Guess we could omit it here yeah :)

File: packages/shared/src/hooks/vote/useFeedVotePost.ts (hunk 7 above)
Comment (rebelchris) [reply packages/shared/src/hooks/vote/useFeedVotePost.ts:46]: Hmm that's true, not sure if it's acceptable or if you have any other ideas on how we can promote it to both mutations optimistically?

Comment (rebelchris) [reply packages/shared/src/hooks/vote/useFeedVotePost.ts:46]: This brings back the initial discussion around actually wanting to do both updates in the same onMutate ideally.
Else one will always be after the mutation happens.

**Review by capJavert — APPROVED** [review]
🚀 awesome!
